src/pages/work/[id]/edit.jsx

```diff
@@ -1,6 +1,38 @@
+import { useEffect, useState } from 'react';
+import { useRouter } from 'next/router';
+import { useUpdateWork } from '@/service/mutations/work';
+import { useGiveUpChallenge } from '@/service/mutations/challenge';
+import WorkForm from '@/components/work/WorkForm';
 import Head from 'next/head';
+import Loader from '@/components/common/Loader';
+import { useGetWork } from '@/service/queries/work';
+import SourceViewer from '@/components/work/SourceViewer';
 
 export default function EditWorkPage() {
+  const [isOpen, setIsOpen] = useState(false);
+  const [content, setContent] = useState('');
+  const router = useRouter();
+  const workId = router.query.id;
+
+  const { data, isPending } = useGetWork(workId);
+  const { mutate: updateWork } = useUpdateWork();
+  const { mutate: giveUpChallenge } = useGiveUpChallenge();
+  console.log(data);
+
+  const challengeId = data?.challenge?.id;
+
+  useEffect(() => {
+    if (data && data.content) {
+      setContent(data.content);
+    }
+  }, [data]);
+
+  const handleUpdateWork = () => {
+    updateWork({ id: workId, data: { content } });
+  };
+
+  if (isPending) return <Loader />;
```

Comment:
```suggestion
 if (isPending) return <Loader msg="작업물을 불러오는 중" />;
```
로딩 메시지를 구체적으로 전달하여 사용자에게 현재 어떤 상황인지 알려주는 것은 어떨까요 👀

Reply:
넵 수정하겠습니다!